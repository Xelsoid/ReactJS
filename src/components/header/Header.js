import PropTypes from "prop-types";
import React from 'react';
import {connect} from "react-redux";

import './header.scss';
import Logo from '../logo/Logo';
import Search from '../search/Search';
import Button from '../button/Button';
import {fetchData} from '../../actions/fetchData'

class Header extends React.Component {
  constructor() {
    super();
    this.state = {
      baseURL: 'https://reactjs-cdp.herokuapp.com/movies',
      searchBy: null,
      search: 'batman',
    };
  }

  concatenateURL = () => {
    const { baseURL, searchBy, search } = this.state;
    const url = `${baseURL}?search=${search}&searchBy=${searchBy}`;
    return url;
  };

  collectInputData = (evt) => {
    this.setState({search: evt.target.value})
  };

  render() {
    const { makeACall } = this.props;
    const { search } = this.state;
    const url = this.concatenateURL();
    return (
      <header className="header">
        <Logo />
        <Search
          value={search}
          callback={this.collectInputData}
        />
        <Button
          id='btnSearch'
          title='Search'
          disabled={false}
          btnClass='btn--primary'
          callback={() => {
            makeACall(url)
          }}
        />
        <div>
          <span>Searched by:</span>
          <Button
            id='btnTitle'
            title='Title'
            disabled={false}
            btnClass='btn--primary'
            callback={() => {this.setState({searchBy: 'title'})}}
          />
          <Button
            id='btnGenre'
            title='Genre'
            disabled={false}
            btnClass='btn--primary'
            callback={() => {this.setState({searchBy: 'genre'})}}
          />
        </div>
      </header>
    )
  }
}

function mapStateToProps(state) {
  return {
    films: state.films
  };
}

function matchDispatchToProps(dispatch) {
  return {
    makeACall: url => dispatch(fetchData(url))
  };
}

Header.defaultProps = {
  makeACall: null,
  search: '',
};

Header.propTypes = {
  makeACall: PropTypes.func,
  search: PropTypes.string,
};


export default connect(mapStateToProps, matchDispatchToProps)(Header);
