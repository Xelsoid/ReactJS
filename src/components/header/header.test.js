import React from 'react';
import Header from './Header';

describe('Header', () => {
  it('renders correctly', () => {
    const component = mount(
      <Header />
    );
    expect(component).toMatchSnapshot();
    expect(component.find('.search-box__wrapper')).toHaveLength(1);
    expect(component.find('.logo')).toHaveLength(1);
  });
});
