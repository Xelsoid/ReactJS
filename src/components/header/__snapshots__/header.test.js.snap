// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<Header
  callback={null}
>
  <header
    className="header"
  >
    <Logo>
      <h1
        className="logo"
      >
        <a
          className="logo__link"
          href="http:/google.com"
        >
          Movie Finder
        </a>
      </h1>
    </Logo>
    <Search>
      <div
        className="search-box__wrapper"
      >
        <label
          className="search-box__search-label"
          htmlFor="searchFilm"
        >
          <i
            aria-hidden="true"
            className="search-box__search-icon fa fa-search"
          />
        </label>
        <input
          className="search-box__search-input"
          id="searchFilm"
          onClick={[Function]}
          placeholder="search"
          type="search"
        />
      </div>
    </Search>
    <Button
      btnClass="btn--primary"
      callback={null}
      disabled={false}
      id="btnSearch"
      title="Search"
    >
      <button
        className="btn btn--primary"
        disabled={false}
        id="Search"
        onClick={[Function]}
        type="button"
      >
        Search
         
        notclicked
      </button>
    </Button>
    <div>
      <span>
        Searched by:
      </span>
      <Button
        btnClass="btn--primary"
        callback={null}
        disabled={false}
        id="btnTitle"
        title="Title"
      >
        <button
          className="btn btn--primary"
          disabled={false}
          id="Title"
          onClick={[Function]}
          type="button"
        >
          Title
           
          notclicked
        </button>
      </Button>
      <Button
        btnClass="btn--primary"
        callback={null}
        disabled={false}
        id="btnGenre"
        title="Genre"
      >
        <button
          className="btn btn--primary"
          disabled={false}
          id="Genre"
          onClick={[Function]}
          type="button"
        >
          Genre
           
          notclicked
        </button>
      </Button>
    </div>
  </header>
</Header>
`;
